extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -0,0 +1,422 @@
+import {CSS} from '../../../build/amp-image-slider-0.1.css';
+import {CommonSignals} from '../../../src/common-signals';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {numeric} from '../../../src/transition';
+import {Platform} from '../../../src/service/platform-impl';
+import {setStyle} from '../../../src/style';
+
+// event-helper.js -> listen; option: passive = true
+// mount both mouse & touch listener ()
+// Test: mainly describes.integration
+// Visual diffing tests
+
+// this.mutateElement(callback), in which move children
+
+
+export class AmpImageSlider extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    const platform_ = new Platform(this.win);
+    /** @private {boolean} */
+    this.isMobile_ = platform_.isAndroid() || platform_.isIos();
+
+    /** @private {boolean} */
+    this.isFollowSlider_ = (this.element.getAttribute('type') === 'follow')
+        && !this.isMobile_; // coerce to drag slider on mobile
+
+    /** @private {!Element} */
+    this.container_ = this.win.document.createElement('div');
+
+    /** @private {?Element} */
+    this.rightAmpImage_ = null;
+
+    /** @private {?Element} */
+    this.leftAmpImage_ = null;
+
+    /** @private {!Element} */
+    this.mask_ = this.win.document.createElement('div');
+
+    /** @private {!Element} */
+    this.bar_ = this.win.document.createElement('div');
+
+    /** @private {!Element} */
+    this.barStick_ = this.win.document.createElement('div');
+
+    /** @private {?Element} */
+    this.barButton_ = null;
+
+    this.handleHover = this.handleHover.bind(this);
+    this.handleClickImage = this.handleClickImage.bind(this);
+    this.handleTapImage = this.handleTapImage.bind(this);
+
+    this.dragStart = this.dragStart.bind(this);
+    this.dragMove = this.dragMove.bind(this);
+    this.dragEnd = this.dragEnd.bind(this);
+
+    this.touchStart = this.touchStart.bind(this);
+    this.touchMove = this.touchMove.bind(this);
+    this.touchEnd = this.touchEnd.bind(this);
+
+    this.pointerMoveX_ = this.pointerMoveX_.bind(this);
+
+    this.vsync_ = this.getVsync();
+
+    this.moveOffset_ = 0;
+    this.splitOffset_ = 0;
+
+    this.isMouseButtonDown_ = false;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.container_.classList.add('i-amphtml-image-slider-container');
+    this.element.appendChild(this.container_);
+
+    const children = this.getRealChildren();
+
+    // at least 2 children
+    if (children.length < 2) {
+      return null;
+    }
+
+    // TODO(kqian): check children type
+    this.leftAmpImage_ = children[0];
+    this.rightAmpImage_ = children[1];
+
+    this.buildImages();
+    this.buildBar();
+  }
+
+  /**
+   * Build image structures
+   */
+  buildImages() {
+    // Hierarchy:
+    // leftMask
+    //   |_ leftAmpImage
+    // rightAmpImage
+    this.container_.appendChild(this.mask_);
```

Comment:
Try putting these in a `mutateElement` block and see if that removes the necessity for `setAsOwner`.

Reply:
Done